Automattic/jetpack
# Plugins page: Adding checks for the deactivation link

Fixes #26307

#### Changes proposed in this Pull Request:

* This PR adds some conditional checks within `const PluginDeactivation` in plugin-deactivation.jsx, to make sure DOM manipulation only happens if the 'Deactivate' DOM element exists. This doesn't exist on sub-sites within multisite networks where Jetpack is network activated, resulting in console errors.
* It also adds a check in class.jetpack.php so make sure we only enqueue the deactivation dialog scripts if the site isn't a multisite with Jetpack network-enabled (and we're not on the main network admin plugins page). The JSX file additions were still needed as on deactivation from the network admin page the same original console errors would appear.

#### Other information:


#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Create a multisite with Jurassic Ninja specialops, network activate Jetpack, then from My Sites > Network Admin > Sites create a new site. Click on 'Setup Jetpack 'and authorize the connection.
* Open the plugins page at wp-admin/plugins.php for the sub-site, and this console errors appears: `react-dom.min.js?ver=17.0.1:141 TypeError: Cannot read properties of null (reading 'setAttribute') at plugins-page.js?minify=false&ver=0236a4b1a47865236a01:2:55999`.
* Apply this PR on a local development site and copy over `const PluginDeactivation` in the generated `plugins-page.js` file (after re-building) (in `_inc/build`), to `plugins-page.js` on the Jurassic Ninja site.
* You'd also need to copy over the changes in `class.jetpack.php`.
* Test on a regular local development site to make sure no errors have been introduced (I've noticed there are other console errors that appear on deactivation which aren't related here, see this PR for more: https://github.com/Automattic/jetpack/pull/26340.).
